### PR TITLE
RFR: added read, edit/update and delete functionality to Sitcom class

### PR DESCRIPTION
**Pull request by RBIII**

### CODE REVIEW

**Comment by dtnorris** [comment]
It looks like 'rspec ./spec/features/create_review_spec.rb' has failures. Please fix and resubmit.

You can see that there are failures when you noticed that Travis CI build failed.

File: app/controllers/sitcoms_controller.rb [hunk 1]
```diff
@@ -1,8 +1,11 @@
 class SitcomsController < ApplicationController
-  before_action :authenticate_user!, only: [:new, :create]
+  before_action :authenticate_user!, only:
+  [:new, :create, :edit, :update, :destroy]
+  before_action :fetch_sitcom, only: [:show, :edit, :destroy]
 
   def index
-    @sitcoms = Sitcom.all.limit(20)
+    sitcoms = Sitcom.all.limit(20)
+    @sitcoms = sitcoms.sort_by { |sitcom| sitcom.start_year }.reverse
```
Comment (SpencerCDixon) [inline app/controllers/sitcoms_controller.rb]: This should be refactored.  Try something like this:

``` ruby
@sitcoms = Sitcom.order(start_year: :desc)
```

http://api.rubyonrails.org/classes/ActiveRecord/QueryMethods.html#method-i-order

File: app/models/sitcom.rb [hunk 2]
```diff
@@ -1,11 +1,16 @@
 class Sitcom < ActiveRecord::Base
   belongs_to :user
-  validates :user_id, presence: true
-  validates :name, presence: true
+  has_many :reviews
+  validates :user_id,
+  presence: true
```
Comment (SpencerCDixon) [inline app/models/sitcom.rb]: these should be indented 2 spaces (the presecnes)

File: app/views/sitcoms/index.html.erb [hunk 3]
```diff
@@ -2,4 +2,11 @@
   <div class="large-12">
     <h1>Sitcoms</h1>
 
+    <ul>
+      <% @sitcoms.each do |sitcom| %>
+        <li><%= link_to (sitcom) do %><%= sitcom.name %></li>
+        <% end %>
```
Comment (SpencerCDixon) [inline app/views/sitcoms/index.html.erb]: this link_to needs to be fixed:
`link_to sitcom.name, sitcom_path(sitcom)`

**Pull request merged**